### ISSUE TITLE
v4, still no luck

### ISSUE THREAD

**Issue by ErikKrause** [asamuzaK/withExEditor#46]
I have updated the extension as well as the host. Settings dialog says:
Host Connection: Connected
Host Version: Compatible 
but "View Source" still opens in the internal viewer.
FF 57.0b14 (64-Bit)

**Comment by asamuzaK** [comment]
This extension does nothing with default "View Source" context menu.
Isn't there "View Source with [Editor Name]" in context menu?

**Comment by ErikKrause** [comment]
No, nothing in the context menu.

**Comment by asamuzaK** [comment]
If possible, can you show me content of editorconfig.json?

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
That's the content of editorconfig.json:
`{
  "editorPath": "C:\\Program Files\\TextPad 7\\TextPad.exe",
  "cmdArgs": [
    "-m",
    "-q"
  ],
  "fileAfterCmdArgs": true
}`
I have now a context menu entry "View Source with withExEditor" after I powered up the computer and started FF (perhaps should have done this yesterday). However, nothing happens if I click it. Setting dialog "Rename Editor Label" is still disabled.

**Comment by ErikKrause** [comment]
Sorry for accidental close.

**Comment by asamuzaK** [comment]
Released a host for debugging. [Release v2.0.2-a.0](https://github.com/asamuzaK/withExEditorHost/releases/tag/v2.0.2-a.0 "Release v2.0.2-a.0 · asamuzaK/withExEditorHost")

* install the host
* restart Firefox (or restart withExEditor from options page)
* open browser console (Ctrl + Shift + J)
* do you see something like below?
  `withexeditorhost: Debug: Found path\to\editorconfig.json`
  `withexeditorhost: Debug: {resEditorConfig: {editorName:...}}`

**Comment by ErikKrause** [comment]
I copied the new host files to the same folder, then ran `node setup.js` again. I hope this is sufficient. Content of editorconfig.json didn't change.
The only line mentioning withExEdit is 
`withexeditorhost: Failed to handle C:\MeineProgramme\ExWith\config\firefox\editorconfig.json.`
after I pressed "Reload" on options page.

**Comment by asamuzaK** [comment]
Got the reason.
You need to save `ExWith` folder under your home directory.
`C:\Users\[YourName]`
not `C:\MeineProgramme\`

Open cmd.exe and run setup.js like below
```
> cd path/to/withExEditorHost
> node setup.js --config-path="C:\Users\[YourName]\ExWith" --browser=firefox
```

**Comment by asamuzaK** [comment]
Plus one more.
<del>You need to save withExEditorHost under your home directory too.</del>
This was not true, but it's better to save host under your home directory.

**Comment by rtoy** [comment]
I have the same problem with both firefox and chrome on a linux system.  Console warning is
```
withexeditorhost: Failed to handle /nfs/apps/src/github/withExEditorHost/config/chrome/editorconfig.json.
```

Everything was working fine until withExEditor auto-updated to 4.0.0. I installed host version v2.0.2-a.0 as well.﻿  There aren't any other messages.

When I actually try to edit something (via context menu), there's the message:
```
withexeditorhost: Application is not executable.
￼```

**Comment by asamuzaK** [comment]
I've removed os.homedir() check.
Try [Release v2.0.2-a.1 · asamuzaK/withExEditorHost](https://github.com/asamuzaK/withExEditorHost/releases/tag/v2.0.2-a.1 "Release v2.0.2-a.1 · asamuzaK/withExEditorHost")

**Comment by rtoy** [comment]
That works great! I'm using it right now to enter this comment.
Thanks for the quick fix. 

Should it be in my home directory, though?

**Comment by asamuzaK** [comment]
Host can be global, but I think at least for the config files, it will be safer to save them under your home directory.

On next release of my host helper module, I am planning to create config files under user's home directory by default.
[Save config files under user's home directory by default · Issue #12 · asamuzaK/webExtNativeMsg](https://github.com/asamuzaK/webExtNativeMsg/issues/12 "Save config files under user's home directory by default · Issue #12 · asamuzaK/webExtNativeMsg")

On Linux, it will be `~/.config/withexeditorhost/config`.

If you want to do that before the next release, run setup.js like below.
`node setup.js --config-path="~/.config/withexeditorhost"`

**Comment by rtoy** [comment]
Yes, that makes sense. I'll try it out soon.

**Comment by asamuzaK** [comment]
@rtoy

If you're still here, would you mind trying [Release v2.1.0-a.1 · asamuzaK/withExEditorHost](https://github.com/asamuzaK/withExEditorHost/releases/tag/v2.1.0-a.1 "Release v2.1.0-a.1 · asamuzaK/withExEditorHost")?

Execute
```
node setup.js
```

No `--config-path` option needed.
It will save config files under user's home directory.
And it will try to remove old config files too, but it will fail if firefox or chrome is still running, so make sure to close both of them before executing setup.js.

Thanks.

**Comment by rtoy** [comment]
Tried it out with firefox (after quitting firefox) and it seems to
work fine.

I'll try with chrome later and let you know.

**Comment by asamuzaK** [comment]
Thanks!

**Comment by asamuzaK** [comment]
@ErikKrause 
Still in trouble?

**Comment by rtoy** [comment]
Tried it out with chrome too.  Works great!

**Comment by asamuzaK** [comment]
Thanks a lot!

**Comment by asamuzaK** [comment]
Released [Release v2.1.0 · asamuzaK/withExEditorHost](https://github.com/asamuzaK/withExEditorHost/releases/tag/v2.1.0 "Release v2.1.0 · asamuzaK/withExEditorHost")

**Comment by asamuzaK** [comment]
I will close this issue.
If you still have trouble, feel free to reopen.
Thanks.
